dtcw

```diff
@@ -6,84 +6,157 @@ set -e
+        # User enforced environment
+        environment=${1}
+        shift 1
+        assert_argument_exists "$@"
     else
-        DTC_PROJECT_BRANCH="-"
+        environment=$(get_environment "$@")
+    fi
+    echo "Using environment: ${environment}"
+
+    if [ "${1}" = "install" ]; then
+        install_component_and_exit "${environment}" "${2-}"
+    elif [ "${1}" = "getJava" ]; then
+        # TODO: remove getJava in the next major release
+        echo "Warning: 'getJava' is deprecated and and will be removed. Use './dtcw install java' instead."
+        install_component_and_exit "${environment}" "java"
     fi
-    export DTC_PROJECT_BRANCH
-fi
 
-# If DTC_HEADLESS provided (from tests), don't change it.
-if [ -z ${DTC_HEADLESS+x} ]; then
-    if [ -t 0 ]; then
-        DTC_HEADLESS=false
-    else
-        echo "Using headless mode since there is no (terminal) interaction possible"
-        DTC_HEADLESS=true
+    # No install command, so forward call to docToolchain but first we check if
+    # everything is there.
+    if [[ ${environment} != docker ]]; then
+        assert_doctoolchain_installed "${environment}" "${DTC_VERSION}"
+        assert_java_version_supported
+
+        # TODO: what if 'doctoolchain' found by $PATH does not match the one from the local environment?
+        # The version provided by $DTC_VERSION could be a different one.
     fi
-fi
-export DTC_HEADLESS
 
-DTC_OPTS="${DTC_OPTS:-} -PmainConfigFile=${MAIN_CONFIG_FILE} --warning-mode=none --no-daemon "
-
-echo "dtcw - docToolchain wrapper V0.38"
-echo "docToolchain V${DTC_VERSION}"
-
-# check if CLI, docker or sdkman are installed
-cli=false
-docker=false
-sdkman=false
-homefolder=false
-java=false
-doJavaCheck=true
-arch=$(uname -m)
-os=$(uname -s)
-
-if command -v doctoolchain &>/dev/null; then
-    echo "docToolchain as CLI available"
-    cli=true
-fi
-if command -v docker &>/dev/null; then
-    echo "docker available"
-    docker=true
-fi
-if command -v sdk &>/dev/null; then
-    echo "sdkman available"
-    sdkman=true
-fi
-if [ -d "${HOME}/.doctoolchain/docToolchain-${DTC_VERSION}/." ]; then
-    echo "home folder exists"
-    homefolder=true
-fi
-if [[ $# -lt 1 ]]; then
-    echo '
-Usage: ./dtcw [option...] [task...]
-
-You can use the same options and tasks as in underlying gradle.
-Use "./dtcw tasks --group doctoolchain" to see available tasks.
-Use "local", "sdk" or "docker" as first argument to force the use of a local, sdkman or docker install.
+    # TODO: can generateDeck, bakePreview be used in combination with other commands?
+    # The code below assumes we have just one task.
+
+    if [ "${1}" = "generateDeck" ]; then
+        # TODO: is this still true? The Dockerfile at
+        # https://github.com/docToolchain/docker-image/blob/ng-beta/alpine/Dockerfile
+        # seems to install reaveal.js also in the docker image.
+        assert_reveal_js_installed "${environment}"
+    fi
+
+    # If bakePreview is called, deactivate daemon.
```

Comment:
this can be removed. The daemon is now always deaktivated since we get some memory leaks when using it

Reply:
I will remove it as soon I'm back from vacation.